0000-settlement-engine/0000-settlement-engine.md

```diff
@@ -0,0 +1,353 @@
+5. If Alice's accounting system triggers any subsequent settlements to Bob, Bob's settlement engine should acknowledge them to the accounting system as normal.
+
+## Specification
+
+Operating a settlement engine requires three components each exposing a particular HTTP API: a settlement engine, an accounting system, and a transport.
+
+API endpoints exposed by these systems may be classified in two ways:
+
+1. Endpoints that may be invoked by one of these other two components, and MUST NOT be invoked manually. At a minimum, all settlement engine implementations MUST support these endpoints.
+2. Endpoints that may be invoked manually by an operator.
+
+This document outlines the first class of endpoints: those only essential to automated interoperability between these systems, and not manual operation.
+
+### Accounts and identifiers
+
+Each account MUST be identified by a unique, URL-safe string.
+
+#### Accounts vs ledger identities
+
+The settlement engine is RECOMMENDED to be responsible for correlating an account identifier to the peer's identity on the shared ledger or settlement system, if required.
+
+For separation of concerns between the accounting system and the settlement system, the accounting system is NOT RECOMMENDED to have knowledge of the peer's identity on the shared settlement system, but rather only the account identifier described here.
+
+#### Multiple accounts
+
+Settlement engine implementations SHOULD support settling with multiple accounts.
+
+#### Correlating messages
+
+The transport MUST correlate incoming messages from the peer to the correct account identifier, and MUST correlate outgoing messages from an account identifier to the correct peer to send them to.
+
+### Units and quantities
+
+#### `Quantity` type
+
+A quantity represents an amount denominated in some unit of a particular fungible asset. In uses of this type, the asset should be implicit.
+
+##### Attributes
+
+- **`amount`** string
+  - Number of the given unit of the asset between 0 and the maximum 64-bit unsigned integer, 18,446,744,073,709,551,615 (inclusive).
+  - The amount is encoded as a string to ensure no precision is lost on platforms that don't natively support full precision 64-bit integers.
+- **`scale`** number
+  - Number of orders of magnitude smaller this unit is as compared to the unit of exchange for the asset.
+  - All units MUST be at least as precise as the unit of exchange. For example, if the asset is U.S. dollars, quantities cannot be denominated in $100 bills, since that's less precise than quantities denominated in multiples of $1. Thus, the scale MUST NOT be a negative integer.
+
+##### Example
+
+To represent $2.54 in units of cents, where the amount is multiple of $0.01:
+
+```json
+{
+  "amount": "254",
+  "scale": 2
+}
+```
+
+#### Choosing scales
+
+The unit or scale used for quantities by the settlement engine is RECOMMENDED to be the smallest denomination of the asset on the settlement system.
```

Comment:
I don't think this recommendation makes a lot of sense; in practice we hardly ever use the same asset scale as the asset scale of the underlying ledger. Usually we go by the smallest "meaningful" amount, whatever that means for the use-case, and then add a few decimal places so that exchange rates have better precision

Reply:
> in practice we hardly ever use the same asset scale as the asset scale of the underlying ledger

I mean for _settlements_, not ILP packets. e.g. XRP payment channel claims are always in drops even if you use scale 9 for ILP packets; ETH payment channel claims are always in wei (scale 18) even if we use scale 9 for ILP packets. Since those units (drops, wei) are the native units of the settlement system, they should be the native units of the SE.

If you want less precision or extra precision for accounting purposes, you can do so, but the idea is that's not a settlement engine concern.